components/plugin_marketplace/marketplace_modal.js

```diff
@@ -86,76 +162,9 @@ export default class MarketplaceModal extends React.Component {
         this.setState({tabKey});
     }
 
-    trackSearch = () => {
-        trackEvent('plugins', 'ui_marketplace_search');
-    }
-
     doSearch = () => {
-        this.trackSearch();
-        this.getMarketplacePlugins();
-    }
-
-    getPluginsListContent = (pluginsArray, installedList) => {
-        if (pluginsArray.length === 0) {
-            let noPluginsMessage = (
-                <FormattedMessage
-                    id='marketplace_modal.no_plugins'
-                    defaultMessage='There are no plugins available at this time.'
-                />);
-            if (installedList) {
-                noPluginsMessage = (
-                    <FormattedMessage
-                        id='marketplace_modal.no_plugins_installed'
-                        defaultMessage='You do not have any plugins installed.'
-                    />);
-            }
-
-            let installLink = null;
-            if (installedList) {
-                installLink = (
-                    <button
-                        className='margin-top x3 style--none color--link'
-                        onClick={() => this.changeTab(MarketplaceTabs.ALL_PLUGINS)}
-                    >
-                        <FormattedMessage
-                            id='marketplace_modal.install_plugins'
-                            defaultMessage='Install Plugins'
-                        />
-                    </button>
-                );
-            }
-
-            return (<div className='no_plugins_div'>
-                <br/>
-                <PluginIcon className='icon__plugin'/>
-                <div className='margin-top x2 light'>
-                    {noPluginsMessage}
-                </div>
-                {installLink}
-            </div>);
-        }
-
-        return (<div className='more-modal__list'>
-            {
-                pluginsArray.map((p) => {
-                    return (
-                        <MarketplaceItem
-                            key={p.manifest.id}
-                            id={p.manifest.id}
-                            name={p.manifest.name}
-                            description={p.manifest.description}
-                            version={p.manifest.version}
-                            isPrepackaged={false}
-                            downloadUrl={p.download_url}
-                            homepageUrl={p.homepage_url}
-                            iconData={p.icon_data}
-                            installed={p.installed_version !== ''}
-                            onConfigure={this.close}
-                            onInstalled={this.getMarketplacePlugins}
-                        />);
-                })
-            }
-        </div>);
+        trackEvent('plugins', 'ui_marketplace_search');
+        this.props.actions.filterPlugins(this.refs.filter ? this.refs.filter.value : null);
```

Comment:
Should we handle the returned error here?

Reply:
Yes, we should! Good catch.